File: app/api/src/Astra.Api/Llm/Archetypes/angular-dotnet8/canonical-delphi-angular-widget-crud/backend/tests/WidgetServiceTests.cs
using Demo.WidgetApi.Models;
using Demo.WidgetApi.Services;
using Xunit;

namespace Demo.WidgetApi.Tests;

public sealed class WidgetServiceTests
{
    private static WidgetService NewService() => new(new InMemoryWidgetRecordPort());

    [Fact]
    public void Create_AssignsIdAndStoresWidget()
    {
        var service = NewService();

        var created = service.Create(new NewWidget("Bolt", 10));

        Assert.True(created.Id > 0);
        Assert.Equal("Bolt", created.Name);
        Assert.Equal(10, created.Quantity);
        Assert.Contains(service.List(), w => w.Id == created.Id);
    }

    [Theory]
    [InlineData("")]
    [InlineData("   ")]
    public void Create_RejectsBlankName(string blankName)
    {
        var service = NewService();

        Assert.Throws<ArgumentException>(() => service.Create(new NewWidget(blankName, 1)));
    }

    [Fact]
    public void Delete_RemovesExistingWidget()
    {
        var service = NewService();
        var created = service.Create(new NewWidget("Nut", 5));

        var removed = service.Delete(created.Id);

        Assert.True(removed);
        Assert.DoesNotContain(service.List(), w => w.Id == created.Id);
    }

    [Fact]
    public void Delete_ReturnsFalseForMissingWidget()
    {
        var service = NewService();

        Assert.False(service.Delete(999));
    }
}

File: app/api/src/Astra.Api/Llm/Archetypes/angular-java/canonical-cpp-widget-crud/backend/src/test/java/com/example/widget/WidgetServiceTests.java
package com.example.widget;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

class WidgetServiceTests {

    private static WidgetService newService() {
        return new WidgetService(new InMemoryWidgetRecordPort());
    }

    @Test
    void create_AssignsIdAndStoresWidget() {
        var service = newService();

        var created = service.create(new NewWidgetRequest("Bolt", 10));

        assertTrue(created.id() > 0);
        assertEquals("Bolt", created.name());
        assertEquals(10, created.quantity());
        assertTrue(service.list().stream().anyMatch(w -> w.id() == created.id()));
    }

    @ParameterizedTest
    @ValueSource(strings = {"", "   "})
    void create_RejectsBlankName(String blankName) {
        var service = newService();

        assertThrows(IllegalArgumentException.class,
            () -> service.create(new NewWidgetRequest(blankName, 1)));
    }

    @Test
    void delete_RemovesExistingWidget() {
        var service = newService();
        var created = service.create(new NewWidgetRequest("Nut", 5));

        var removed = service.delete(created.id());

        assertTrue(removed);
        assertFalse(service.list().stream().anyMatch(w -> w.id() == created.id()));
    }

    @Test
    void delete_ReturnsFalseForMissingWidget() {
        var service = newService();

        assertFalse(service.delete(999));
    }
}
